src/backend/src/utils/projects.utils.ts

```diff
@@ -304,3 +304,44 @@ export const checkMaterialInputs = async (
     if (!unit) throw new NotFoundException('Unit', unitName);
   }
 };
+
+export const validateBlockedBys = async (blockedBy: WbsNumber[]) => {
+  blockedBy.forEach((dep: WbsNumber) => {
+    if (dep.workPackageNumber === 0) {
+      throw new HttpException(400, 'A Project cannot be a Blocker');
+    }
+  });
+
+  const blockedByWBSElems: (WBS_Element | null)[] = await Promise.all(
+    blockedBy.map(async (ele: WbsNumber) => {
+      return await prisma.wBS_Element.findUnique({
+        where: {
+          wbsNumber: {
+            carNumber: ele.carNumber,
+            projectNumber: ele.projectNumber,
+            workPackageNumber: ele.workPackageNumber
+          }
+        }
+      });
+    })
+  );
+
+  const blockedByIds: number[] = [];
+  // populate blockedByIds with the element ID's
+  // and return error 400 if any elems are null
+
+  let blockedByHasNulls = false;
+  blockedByWBSElems.forEach((elem) => {
+    if (!elem) {
+      blockedByHasNulls = true;
+    } else {
+      blockedByIds.push(elem.wbsElementId);
+    }
+  });
+
+  if (blockedByHasNulls) {
+    throw new HttpException(400, 'One of the blockers was not found.');
```

Comment:
This exception could be in the `forEach` to avoid the variable

Adding to `blockedByIds` can stay in the `forEach` for a single-pass approach or could be mapped from `blockedByWBSElems`